src/conversions/array.rs

```diff
@@ -149,17 +174,69 @@ mod min_const_generics {
 #[cfg(not(min_const_generics))]
```

Comment:
Unnngh just another year until we bump msrv and throw this one away 😅. Nicely done for figuring out the `ArrayGuard` technique!